src/NLog/Internal/FileAppenders/FileAppenderCache.cs

```diff
@@ -255,6 +296,21 @@ public FileCharacteristics GetFileCharacteristics(string fileName)
         }
 
         /// <summary>
+        /// Invalidates appenders for all files that were archived.
+        /// </summary>
+        public void InvalidateAppendersForInvalidFiles()
+        {
+#if !SILVERLIGHT && !__IOS__ && !__ANDROID__
```

Comment:
I think we should not expose this method for `f !SILVERLIGHT && !__IOS__ && !__ANDROID__`. What do you think